lib/Parse/ParseType.cpp

```diff
@@ -533,9 +530,8 @@ ParserResult<TypeRepr> Parser::parseType(Diag<> MessageID,
           if (!argTy.getPtrOrNull())
             return false;
           SubsTypesVec.push_back(argTy.get());
-          if (consumeIf(tok::comma))
-            continue;
-          break;
+          if (!consumeIf(tok::comma))
```

Comment:
These hanging breaks make me nervous because it makes the flow of control less obvious. Perhaps these needs to be `do-while` loops instead.

Reply:
ElementContext is a RAII object and it needs consumeIf(tok::comma) within the lifetime of it.

Reply:
So that is not possible. Sorry @CodaFi